includes/class-amp-theme-support.php

```diff
@@ -370,7 +392,37 @@ public static function get_theme_support_args() {
 	 * @since 0.7
 	 */
 	public static function finish_init() {
-		if ( ! is_amp_endpoint() ) {
+		if ( self::is_standalone_content_request() ) {
+
+			// Fail request if it is not for a single post that supports AMP.
+			$queried_object = get_queried_object();
+			if ( ! ( $queried_object instanceof WP_Post ) || ! post_supports_amp( $queried_object ) ) {
+				wp_die(
+					esc_html__( 'AMP content is not available for this URL.', 'amp' ),
+					esc_html__( 'AMP Content Unavailable', 'amp' ),
+					[ 'response' => 400 ]
+				);
+			}
+
+			// Prevent including extraneous stylesheets.
+			add_filter( 'show_admin_bar', '__return_false' );
+			add_filter(
+				'print_styles_array',
+				function ( $handles ) {
+					return array_intersect( $handles, [ 'amp-default', 'wp-block-library', 'wp-block-library-theme' ] );
```

Comment:
Should we allow other styles to be included in this list? Perhaps the list of styles should be filterable so that a site can decide whether or not additional styles should be included?

This is a similar issue for stories, where it should be possible for a site to opt-in to an additional stylesheet to be included.

https://github.com/ampproject/amp-wp/blob/186c83ee4d8b0fc7d383ae27fd2891c9bb094ec7/includes/templates/single-amp_story.php#L72-L75

https://github.com/ampproject/amp-wp/blob/186c83ee4d8b0fc7d383ae27fd2891c9bb094ec7/includes/class-amp-story-post-type.php#L677-L711